lib/redaction.js

```diff
@@ -81,8 +81,19 @@ function redaction (opts, serialize) {
 
   return [...Object.keys(shape), ...Object.getOwnPropertySymbols(shape)].reduce((o, k) => {
     // top level key:
-    if (shape[k] === null) o[k] = topCensor
-    else o[k] = fastRedact({ paths: shape[k], censor, serialize, strict })
+    if (shape[k] === null) {
+      o[k] = topCensor
+    } else {
+      const wrappedCensor = typeof censor === 'function' ? (value, path) => {
+        return censor(value, [k, ...path])
+      } : censor
```

Comment:
Why is this needed? I would prefer if we avoid this.

Reply:
Answering for this and also quoted comment
> Can you please this? I don't understand the change.

This PR https://github.com/davidmarkclements/fast-redact/pull/16 added the functionality to pass a `path` array to the `censor` function.

We had to do this because otherwise the `path` array is missing the top level key.

For instance, say we have a logger set up like so:

```js
const pino = require('.')
const logger = pino({redact: {paths: ['a.b.c'], censor: console.log }})
logger.info({a: {b: {c: 'redact me'}}})
```

this will log out: `redact me ['a', 'b', 'c']`
without this change this will log out: `redact me ['b', 'c']`

This is because, fast-redact is being used per key on the logged object, so it is passed a redaction path of `'b.c'` and then works on the object `{b: {c: 'redact me'}}` instead of `'a.b.c'`, working on the object `{a: {b: {c: 'redact me'}}}`.